server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationStatsIT.java

```diff
@@ -72,6 +72,7 @@ public void testSegmentReplicationStatsResponse() throws Exception {
             assertEquals(segmentReplicationStatsResponse.getReplicationStats().size(), 1);
             assertEquals(segmentReplicationStatsResponse.getTotalShards(), numShards * 2);
             assertEquals(segmentReplicationStatsResponse.getSuccessfulShards(), numShards * 2);
+            assertNotNull(currentReplicationState);
```

Comment:
Asserting not null will change the NPE to an AssertionError. This assertion statement wouldn't change any aspect of the `testSegmentReplicationStatsResponse` test and thus wouldn't fix any flakyness here. May be we need to change the description/PR title here.

We may want to investigate why `currentReplicationState` is null and fix it.

Reply:
Can you share corresponding gradle run where this failure was observed ? We can use the seed to reproduce this scenario and identify why `currentReplicationState` is null

Reply:
I see this code is run inside an `assertBusy` block (needed to expand changes here :) ) which is ran for `1 minute`. So this assertion would be caught and this logic will be retried. Please ignore my previous comment.